sources/ghostdog/src/main.rs

```diff
@@ -160,6 +175,51 @@ fn is_neuron_attached() -> Result<()> {
     }
 }
 
+fn is_infiniband_attached() -> Result<()> {
+    // confirm device has `SW_MNG` in the VPD file
+    let infiniband_sys_devices =
+        fs::read_dir("/sys/class/infiniband/").context(error::InfinibandSysDevicesSnafu)?;
+
+    for device in infiniband_sys_devices {
+        let device = device.context(error::InfinibandSysDevicesSnafu)?;
+        let device_name: OsString = device.file_name().clone();
+        let vpd_file = Path::new(&device.path()).join("device").join("vpd");
+        if vpd_file.exists() {
+            let vpd_file_bytes: &[u8] =
+                &std::fs::read(vpd_file.as_path()).context(error::ReadFileSnafu {
+                    path: vpd_file.as_path(),
+                })?;
+            let vpd_file_string = String::from_utf8_lossy(vpd_file_bytes);
+            if !vpd_file_string.contains("SW_MNG") {
+                continue;
+            }
+        }
```

Comment:
My main misgiving about this is that the command is very generic - `ghostdog infiniband-present` - but it ends up being super specific to a particular configuration of an NVIDIA device.

I would factor this into two parts:
1. generic function that can iterate across Infiniband device entries and check some files for properties
2. specific subcommand for the task, such as `ghostdog infiniband-sw-mng`

Reply:
I've found a possible different path which involves passing the Port GUID to each process via `-g` on their command line invocation. This can be sourced from an `EnvironmentFile` which allows a much simpler interface where `ghostdog` only needs to write to an environment file if there is a suitable device present, and if not, then systemd will do the right thing and skip the right services. This means `ghostdog` will have an argument like `ghostdog write-infiniband-sw-mng-guid` which is more accurate and targeted at writing this file.

Reply:
I switched to the writing the environment file. `ghostdog` takes this file as an argument now but the last comment remains the same. I'll update the description to reflect the new approach.